pkg/middleware/auth.go

```diff
@@ -0,0 +1,65 @@
+package middleware
+
+import (
+	"net/http"
+
+	clouddriver "github.com/billiford/go-clouddriver/pkg"
+	"github.com/billiford/go-clouddriver/pkg/fiat"
+	"github.com/gin-gonic/gin"
+)
+
+//authApplication takes a list of permissions
+//authAccount
+//go:generate counterfeiter . Auth
+type Auth interface {
```

Comment:
There shouldn't be an interface for the middleware - we can call it directly. The only thing that needs an interface right now is the fiat client.